implementation/jobberknoll/core/shared/option.ts

```diff
@@ -0,0 +1,12 @@
+export type Some<T> = { tag: "some"; value: T };
+export type None = { tag: "none" };
+
+export type Option<T> = Some<T> | None;
+
+export const some = <T>(value: T): Some<T> => ({ tag: "some", value });
+export const none = (): None => ({ tag: "none" });
+
+export const isSome = <T>(option: Option<T>): option is Some<T> =>
+  option.tag === "some";
+export const isNone = <T>(option: Option<T>): option is None =>
+  option.tag === "none";
```

Comment:
Perhaps you should do it in rust after all? xD

Reply:
I should have 😭

Reply:
I understand your pain, throwing exceptions is the worst thing I do in my service implementation, but in my case it's standardized enough that I don't undertake to do everything through types.

Reply:
@jakubzehner .NET 8.0 solved it nicely:
- https://nikiforovall.github.io/dotnet/2024/08/17/result-endpoints.html
- https://fast-endpoints.com/docs/get-started#union-type-returning-handler

We will see it in Spring too, probably in 5-10 years.